default_i8_delegates.rb

```diff
@@ -119,13 +119,15 @@ def _header_value
   # Fetch the URL using the HEAD method.
   #
   # Adapted from https://stackoverflow.com/a/6934503
-  def _fetch(uri, limit = 10)
+  def _fetch(uri, limit = 10, last_response = nil)
     # StandardError should suffice.
     raise 'HTTP redirect too deep' if limit == 0
 
     head = Net::HTTP::Head.new uri
-    _headers.each do |header, value|
-      head[header] = value
+    if last_response == nil or last_response['x-dgi_i8_helper_flysystem_s3__no_headers'] != 'true'
+      _headers.each do |header, value|
+        head[header] = value
+      end
     end
```

Comment:
Just for reference, PR on the other side defining this header:  https://github.com/discoverygarden/dgi_i8_helper/pull/218/files#diff-6f025752c8c25ccfd7692b28db002f71b2c8cae55c528c66a9959050e485103aR150-R155